### PR TITLE
Speeding up - Using Spark

### PR DESCRIPTION
**Pull request by dav009**
Transforming a wiki to jsonwiki TAKES AGES. Literally one day. Running in one core.

This PR makes jsonpedia work with spark.

- assumes the input wikipedia is uncompressed. (yes those gz, bz files won't work)
- instead of calling the CLI, to use it we just spark-submit the CLI class 

`bin/spark-submit --driver-memory 10G --class it.cnr.isti.hpc.wikipedia.cli.MediawikiToJsonCLI  /Users/dav009/source/idio-jsonwiki/json-wikipedia/target/json-wikipedia-1.0.0-jar-with-dependencies.jar -input /Users/dav009/enwiki-20150403-small -output ~/output_spark/new_jsonpedia -lang en`

tested this locally

### CODE REVIEW

**Comment by keynmol** [comment]
How big is the speedup?

**Comment by dav009** [comment]
Nothing fancy, just running things once in my laptop:
```
decompressed (small wiki: `enwiki-20150403-small.tar.gz`)
spark: 135770 ms 
non-spark: 219084 ms
```

File: README.md [hunk 1]
```diff
@@ -3,26 +3,45 @@ json-wikipedia ![json-wikipedia](https://dl.dropboxusercontent.com/u/4663256/tmp
...
+
+
+## Convert the Wikipedia XML to JSON
+
+### The Docker way
+
+Enjoy a dockerized image:
 
-compile the project running 
+   `docker run -v <LOCALPATH>:/mnt -i -t dav009/jsonpedia -input <PATHTOWIKI> -output <OUTPUTPATH> -lang <LANG>`
```
Comment (keynmol) [inline README.md]: isn't it [dav009/jsonwikipedia](https://registry.hub.docker.com/u/dav009/jsonwikipedia/) ?

**Comment by Lugrin** [comment]
There is a mix of tabs and spaces. It's probably worth converting all tabs into spaces. I suppose IntelliJ can do that in 3 clicks. If you do so, please open a new review so that your changes so far don't get buried.

**Comment by dav009** [comment]
I think that will change the whole project
I will get that out in other PR.

**Comment by dav009** [comment]
This is because the project use tabs, but intelliJ by default uses spaces.
So whenever I touch anything from the previous source code, intelliJ will use spaces whenever I put a tab.

**Comment by dav009** [comment]
?

**Comment by dav009** [comment]
@keynmol sorry can you comment on current commiits? 
Otherwise it makes it harder to see on what line you are dropping your comment on, or whether you are replying to a previous message

**Comment by keynmol** [comment]
I thought that's exactly what I was doing... weird

File: README.md [hunk 2]
```diff
@@ -3,26 +3,45 @@ json-wikipedia ![json-wikipedia](https://dl.dropboxusercontent.com/u/4663256/tmp
...
+5. do:
+
+    SPARKFOLDER/bin/spark-submit --driver-memory 10G --class it.cnr.isti.hpc.wikipedia.cli.MediawikiToJsonCLI json-wikipedia-1.0.0-jar-with-dependencies.jar -input <PATHTODBPEDIADUMP> -output <PATHTONEWJSONPEDIA> -lang <LANG>
+
+this produces in `<PATHTONEWJSONPEDIA>` the JSON version of the dump
+
+### How does Jsonpedia look like?
+
+ ([here you can find an example](https://dl.dropboxusercontent.com/u/4663256/tmp/json-wikipedia-sample.json)). Each line of the file contains an article
+of dump encoded in JSON. Each JSON line can be deserialized in an [Article](http://sassicaia.isti.cnr.it/javadocs/json-wikipedia/it/cnr/isti/hpc/wikipedia/article/Article.html) object,which represents an_enriched_ version of the wikitext page. The Article object contains:
```
Comment (keynmol) [inline README.md]: missing space before `_encriched`

**Comment by keynmol** [comment]
Version bump?

**Comment by dav009** [comment]
bumped

**Pull request merged**